packages/studio/src/components/StringPropInput.tsx

```diff
@@ -47,19 +47,21 @@ export default function StringPropInput({
     [displayValue, onChange]
   );
 
-  const fieldPickerFilter = useCallback(
-    (value: unknown) =>
-      TypeGuards.valueMatchesPropType({ type: PropValueType.string }, value),
-    []
-  );
-
   return (
     <FieldPickerInput
       onInputChange={handleChangeEvent}
       handleFieldSelection={appendField}
       displayValue={displayValue}
-      fieldFilter={fieldPickerFilter}
+      fieldFilter={useFieldPickerFilter()}
       disabled={disabled}
     />
   );
 }
+
+function useFieldPickerFilter() {
+  return useCallback(
```

Comment:
since this function is outside the component, it doesn't need the useCallback (correct me if im wrong!)

Reply:
oooh u right